src/xss-filters.js

```diff
@@ -80,6 +82,29 @@ exports._getPrivFilters = function () {
             return String.fromCharCode((codePoint >> 10) + 0xD800, (codePoint % 0x400) + 0xDC00);
         };
 
+    // patch document.write() and document.writeln() to properly handle NULL for IE 9 or below
```

Comment:
this polyfill targets only those IE7-9, where string after a NULL char will be skipped. other browsers are not affected.
It basically wraps the original `document.write()`, so its original unknown implementation is kept intact.
Before passing over to the original function, the only thing it does here is just replace NULL with \uFFFD.
Therefore, all chars will be consumed for printing. Let's further discuss when we meet. :)